src/components/SubscriptionTable/SubscriptionTable.tsx

```diff
@@ -0,0 +1,233 @@
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number>(0);
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc'>('asc');
+  const [searchValue, setSearchValue] = React.useState('');
+
+  const columnNames = {
+    number: 'Subscription number',
+    contractNumber: 'Contract number',
+    quantity: 'Subscription quantity',
+    startDate: 'Start date',
+    endDate: 'End date'
+  };
+
+  const getSortParams = (columnIndex: number): ThProps['sort'] => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+      defaultDirection: 'asc'
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex
+  });
+
+  const sortSubscriptions = (
+    subscriptions: TypeCorrectedSubscription[],
+    index: number
+  ): TypeCorrectedSubscription[] => {
+    const keys = [
+      'number',
+      'contractNumber',
+      'quantity',
+      'startDate',
+      'endDate'
+    ] as (keyof TypeCorrectedSubscription)[];
+    return subscriptions.sort((a: TypeCorrectedSubscription, b: TypeCorrectedSubscription) => {
+      let result = 0;
+      const aValue = a[keys[index]];
+      const bValue = b[keys[index]];
+
+      if (aValue < bValue) {
+        result = -1;
+      } else if (aValue > bValue) {
+        result = 1;
+      }
+
+      return activeSortDirection == 'asc' ? result : -1 * result;
+    });
+  };
+
+  const filter = (
+    subscriptions: TypeCorrectedSubscription[],
+    search: string
+  ): TypeCorrectedSubscription[] => {
+    search = search.trim();
+    return subscriptions.filter((subscription) => {
+      const n = subscription.number != -1 ? subscription.number.toString() : 'Not Available';
+      const cn =
+        subscription.contractNumber != -1
+          ? subscription.contractNumber.toString()
+          : 'Not Available';
+      return n.includes(search) || cn.includes(search);
+    });
+  };
+
+  const getPage = (subscriptions: TypeCorrectedSubscription[]) => {
+    const first = (page - 1) * perPage;
+    const last = first + perPage;
+    return subscriptions.slice(first, last);
+  };
+
+  const sortedSubscriptions = sortSubscriptions(castData, activeSortIndex);
+  const filteredSubscriptions = filter(sortedSubscriptions, searchValue);
+  const pagedSubscriptions = getPage(filteredSubscriptions);
+
+  const pagination = (variant = PaginationVariant.top) => {
+    return (
+      <Pagination
+        itemCount={subscriptions.length}
+        perPage={perPage}
+        page={page}
+        onSetPage={(_e: React.MouseEvent, p: number) => {
+          setPage(p);
+        }}
+        onPerPageSelect={(_e: React.MouseEvent, p: number) => {
+          setPerPage(p);
+          setPage(1);
+        }}
+        variant={variant}
+      />
+    );
+  };
+
+  return (
+    <>
+      <Card>
+        <CardTitle>Subscription Details</CardTitle>
```

Comment:
2) Maybe instead of having it be a card here in the Subscription Table, it should be a Title component in the DetailsPage, similar to how it's done in the SubscriptionInventoryPage.

Reply:
Good call, this is a better pattern